modules/metadata-service/api-gateway.tf

```diff
@@ -146,7 +157,19 @@ resource "aws_api_gateway_deployment" "this" {
   }
 }
 
-resource "aws_api_gateway_stage" "this" {
+resource "aws_api_gateway_stage" "staging" {
+  deployment_id = aws_api_gateway_deployment.this.id
+  rest_api_id   = aws_api_gateway_rest_api.this.id
+  stage_name    = "staging_${local.api_gateway_stage_name}"
+
+  tags = var.standard_tags
+}
+
+resource "aws_api_gateway_stage" "prod" {
+  # depends_on basically ensures that deployment to prod happens only if deployment to staging goes through
+  depends_on = [
+    aws_api_gateway_stage.staging
+  ]
```

Comment:
doesn't this not exactly solve the issue we are having, because we are still gonna be taking down the production stage each time we redeploy so we will still have some downtime on the api? Like if the metaflow-wrapped VA is pointing to the `api/` stage, we'll still incur some downtime there while that stage updates, even with this new `staging` stage, right? Seems like we still need to build out this workflow a little more to avoid downtime, unless I am missing something

Also nit but I think we should change the `local.api_gateway_stage_name` to a tf variable that can be provided, just to provide some flexibility around the name.

Reply:
Ah yaah good point... that was my thought initially as well when I first started working on this.. but after reading more about API Gateway deployments and a bunch of stackoverflow threads, it seems that updating a stage in API Gateway to point to a new deployment id should happen without downtime (i.e. instantly? or rather requests while the update is happening will go the to old deployment and requests after the update is complete will go to the new deployment). Now, my hypothesis is that the reason this design failed last time is most likely due to the [bug](https://github.com/hashicorp/terraform-provider-aws/issues/22866) in the terraform aws provider version. There is no way for me to confirm this 100% but I believe this is what was happening lol - and I think only testing it out will give us the real answer.  The new `staging` stage will ensure that if there ever happens to be another bug in terraform the `prod` stage won't get affected since the `staging` deployment will fail - it is essentially a way to work around the lack of rollback in terraform.

Reply:
ah alright sounds good. Yeah lets try this out then. Actually, maybe we can just trigger the `apply` locally on the `analytics_metaflow` repo to sanity check that it will go through, if you havent done so already

Reply:
Alright, tested `apply` locally and it went through OK. I guess we will need to do a real test with a VA running "live" (in integration) and try to re-deploy and monitor for errors.